File: src/models/job_log.go
package models

import (
	"time"
)

// JobLog is JobLog
//
//go:generate reform
//reform:job_logs
type JobLog struct {
	ID        int       `reform:"id,pk" json:"id"`
	JobID     int       `reform:"job_id" json:"-"`
	JobPath   string    `reform:"job_path"`
	Content   string    `reform:"content" json:"content"`
	CreatedAt time.Time `reform:"created_at" json:"createdAt"`
}

File: client/src/models/job_log.ts
type Props = {
  id: number
  job_id: number
  content: string
  createdAt: string
}

class JobLog implements Props {
  id: number = 0
  job_id: number = 0
  content: string = ''
  createdAt: string = ''

  constructor(props: Partial<Props>) {
    for(const prop in props) {
      (this as any)[prop] = (props as any)[prop]
    }
  }

  static deserialize(json: any): JobLog {
    const model = new JobLog(json as Partial<Props>)
    return model
  }
}

export default JobLog